crates/storage/src/errors.rs

```diff
@@ -0,0 +1,19 @@
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum StoreError {
+    #[error("Write transaction failure")]
+    Database(#[from] redb::Error),
+
+    #[error("Transaction error")]
+    TransactionError(#[from] redb::TransactionError),
+
+    #[error("Commit error")]
+    CommitError(#[from] redb::CommitError),
+
+    #[error("Storage error")]
+    StorageError(#[from] redb::StorageError),
+
+    #[error("Table error")]
+    TableError(#[from] redb::TableError),
```

Comment:
```suggestion
    #[error("Write transaction failure: {0}")]
    Database(#[from] redb::Error),

    #[error("Transaction error: {0}")]
    TransactionError(#[from] redb::TransactionError),

    #[error("Commit error: {0}")]
    CommitError(#[from] redb::CommitError),

    #[error("Storage error: {0}")]
    StorageError(#[from] redb::StorageError),

    #[error("Table error: {0}")]
    TableError(#[from] redb::TableError),
```

we want to display the underlying error